### ISSUE TITLE
[bug] imports inside the specs failed to resolve when running tests on GitHub Actions

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#677]
### Describe the bug

When upgrading to `@typescript-eslint` v8.0.0, type-aware rules could not be tested on GitHub Actions because import statements in spec code could not be resolved when running tests in a CI environment.

### Reproduction

```ts
{
  code: /* tsx */ `
    /// <reference types="react" />
    /// <reference types="react-dom" />

    import * as React from "react";

    const App: React.FC<{ id: string; className: string }> = (props) => {
        return <div id={props.id} className={props.className} />
    }
  `,
  errors: [
    {
      messageId: "PREFER_READ_ONLY_PROPS",
    },
  ],
}
```

### Expected behavior

Types from import statements are resolved correctly, as `@typescript-eslint` v7 does.

### Platform and versions

```Text
Current runner version: '2.317.0'
Operating System
  Ubuntu
  22.04.4
  LTS
Runner Image
  Image: ubuntu-22.04
  Version: 20240721.1.0
  Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240721.1/images/ubuntu/Ubuntu2204-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240721.1
Runner Image Provisioner
  2.0.373.1

eslint: 9.8.0
@typescript-eslint: 8.0.0
@eslint-react: v1.8.0-next.0
```

### Stack trace

```Text
⎯⎯⎯⎯⎯⎯ Failed Tests 11 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      const App: React.FC<{ readonly id: string; readonly className: string }> = ({ id, className }) => {
          return <div id={id} className={className} />
      }
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 7,
    column: 83,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 7,
    endColumn: 100
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      const defaultProps = { id: "default-id", className: "default-class" } as const;
      type Props = typeof defaultProps;
      const App: React.FC<Props> = ({ id, className }) => {
          return <div id={id} className={className} />
      }
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 9,
    column: 37,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 9,
    endColumn: 54
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      const defaultProps = { id: "default-id", className: "default-class" } as const;
      const App: React.FC<typeof defaultProps> = ({ id, className }) => {
          return <div id={id} className={className} />
      }
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 8,
    column: 51,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 8,
    endColumn: 68
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      type Props = { readonly id: string; readonly className: string };
      const App = React.memo<Props>(({ id, className }) => {
          return <div id={id} className={className} />
      });
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 8,
    column: 38,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 8,
    endColumn: 55
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      const defaultProps = { id: "default-id", className: "default-class" } as const;
      type Props = typeof defaultProps;
      const App = React.memo<Props>(({ id, className }) => {
          return <div id={id} className={className} />
      });
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 9,
    column: 38,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 9,
    endColumn: 55
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      type Props = { readonly id: string; readonly className: string };
      const App = React.forwardRef<HTMLDivElement, Props>(({ id, className }, ref) => {
          return <div id={id} className={className} ref={ref} />
      });
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 8,
    column: 60,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 8,
    endColumn: 77
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      const defaultProps = { id: "default-id", className: "default-class" } as const;
      type Props = typeof defaultProps;
      const App = React.forwardRef<HTMLDivElement, Props>(({ id, className }, ref) => {
          return <div id={id} className={className} ref={ref} />
      });
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 9,
    column: 60,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 9,
    endColumn: 77
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > valid > 
      /// <reference types="react" />
      /// <reference types="react-dom" />

      import * as React from "react";

      type Props = { readonly id: string; readonly className: string };
      const App = React.memo(React.forwardRef<HTMLDivElement, Props>(({ id, className }, ref) => {
          return <div id={id} className={className} ref={ref} />
      }));
    
AssertionError: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/prefer-read-only-props',
    severity: 1,
    message: "A function component's props should be read-only.",
    line: 8,
    column: 71,
    nodeType: 'ObjectPattern',
    messageId: 'PREFER_READ_ONLY_PROPS',
    endLine: 8,
    endColumn: 88
  }
]

- Expected
+ Received

- 0
+ 1

 ❯ testValidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:914:20
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1261:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[8/11]⎯

 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > invalid > 
        /// <reference types="react" />
        /// <reference types="react-dom" />

        import * as React from "react";

        const App: React.FC<{ id: string; className: string }> = (props) => {
            return <div id={props.id} className={props.className} />
        }
      
 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > invalid > 
        /// <reference types="react" />
        /// <reference types="react-dom" />

        import * as React from "react";

        export const App: React.FC<{ id: string; className: string }> = (props) => {
          return <div className={props.className} id={props.id} />
        }
      
 FAIL  packages/plugins/eslint-plugin-react-x/src/rules/prefer-read-only-props.spec.ts > prefer-read-only-props > invalid > 
        /// <reference types="react" />
        /// <reference types="react-dom" />

        import * as React from "react";

        export const App: React.FC<{ readonly id: string; readonly className: string } | { id: string; className: string }> = (props) => {
          return <div className={props.className} id={props.id} />
        }
      
AssertionError: Should have 1 error but had 0: []

- Expected
+ Received

- 1
+ 0

 ❯ testInvalidTemplate node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:998:24
 ❯ node_modules/.pnpm/eslint@9.8.0/node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[9/11]⎯

 Test Files  1 failed | 68 passed (69)
      Tests  11 failed | 2160 passed (2171)
   Start at  18:16:07
   Duration  41.99s (transform 2.97s, setup 0ms, collect 87.32s, tests 18.28s, environment 14ms, prepare 5.83s)

 ELIFECYCLE  Test failed. See above for more details.
Error: Process completed with exit code 1.
```
